lib/tests/CMakeLists.txt

```diff
@@ -29,14 +31,32 @@ endif()
 # described in https://gcc.gnu.org/ml/gcc-help/2015-08/msg00035.html, we get
 # the error undefined reference to symbol '__tls_get_addr@@GLIBC_2.3'.
 # Build mock_server as a separate shared library to avoid this error.
+add_library(pcp-chunks # pcp-chunk is needed by MockServer (no threads)
+    ${SOURCE_DIR}/protocol/chunks.cc
+    ${SOURCE_DIR}/protocol/message.cc
+    ${SOURCE_DIR}/protocol/serialization.cc
+    ${SOURCE_DIR}/validator/schema.cc
+    ${SOURCE_DIR}/validator/validator.cc)
+target_link_libraries(pcp-chunks PRIVATE
+    ${Boost_REGEX_LIBRARY}
+    ${Boost_SYSTEM_LIBRARY}
+    ${LEATHERMAN_LIBRARIES})
+add_dependencies(pcp-chunks valijson)
+
+# Generate the export header and export symbols
+symbol_exports(pcp-chunks "${CMAKE_CURRENT_LIST_DIR}/../inc/cpp-pcp-client/export-pcp-chunks.h")
```

Comment:
Yeah, I thought so.

Right now, I'm just curious to see if this work, as I've been hacking on it for some time (the STATIC approach does not work).

The `symbol_exports()` approach with multiple _EXPORT tokens on headers requires that the export.h files must be rewritten, because you must guarantee that the one of the second library is there before it gets actually generated.